WordPress/Classes/ViewRelated/Media/MediaItemViewController.swift

```diff
@@ -79,10 +79,26 @@ class MediaItemViewController: UITableViewController {
     }
 
     private func reloadViewModel() {
+        guard !isMediaDeleted else {
+            handleDeletedMedia()
+            return
+        }
+
         updateViewModel()
         tableView.reloadData()
     }
 
+    private var isMediaDeleted: Bool {
+        return media.isDeleted || media.managedObjectContext == nil
+    }
+
+    private func handleDeletedMedia() {
```

Comment:
I think it's probably okay to leave it as-is? The view controller gets popped off straight away anyways, and media is currently a constant.